### PR TITLE
Riffa lock

### PR DESCRIPTION
**Pull request by aacuevas**
Performs a lock on the FPGA handle at context init for tiffa so the device resets when it's closed.
Requires open-ephys modified riffa driver version 6.0.1

### CODE REVIEW

File: api/liboni/drivers/riffa/onidriver_riffa.c [hunk 1]
```diff
@@ -64,9 +64,17 @@ oni_driver_ctx oni_driver_create_ctx()
...
+    if (host_idx < 0) host_idx = 0; //we should implement an auto list and select method
     ctx->fpga = fpga_open(host_idx);
     if (ctx->fpga == NULL) return ONI_EINIT;
+    
+    //get a lock so this handle resets the system on close
+    if (riffa_lock(ctx->fpga) != 0) //if not possible to get the lock.
+    {
+        fpga_close(ctx->fpga);
+        ctx->fpga = NULL;
+        return ONI_ERETRIG; //seemed the most appropriate error
```
Comment (jonnew) [inline api/liboni/drivers/riffa/onidriver_riffa.c:76]: I would replace with ONI_EINIT

Comment (jonnew) [reply api/liboni/drivers/riffa/onidriver_riffa.c:76]: I'm also not opposed to creating another error for this. The whole point is to be clear about what is going wrong.

Comment (aacuevas) [reply api/liboni/drivers/riffa/onidriver_riffa.c:76]: Yeah, EINIT seemed too generic, like "device could not init" but in this case "device could not be locked" which is different.

Maybe we should review the error codes. Some are from when the lib was xillybus-only (ONI_ESEEKFAILURE or ONI_EPATHINVALID are good examples)

Comment (jonnew) [reply api/liboni/drivers/riffa/onidriver_riffa.c:76]: Yeah I agree. OK, I will merge this now so it does not sit and start an issue about error codes.

**Pull request merged**